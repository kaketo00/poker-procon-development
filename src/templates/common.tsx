import { Box } from '@chakra-ui/react';
import Head from 'next/head';
import { type PropsWithChildren } from 'react';

import Loading from '@/components/atoms/loading';
import Footer from '@/components/organisms/footer';
import Header from '@/components/organisms/header';
import { SERVICE_NAME } from '@/constants';

interface Props extends PropsWithChildren {
  isLoading?: boolean;
  title?: string;
}

const CommonTemplate = ({ isLoading = false, title, children }: Props) => (
  <Box
    w="full"
    h="100dvh"
    bgImage="url('/bg-common.png')"
    bgSize="cover"
    bgPos="center"
    bgColor="black"
  >
    <>
      {isLoading && <Loading />}
      <Head>
        <title>{title ? `${title} - ${SERVICE_NAME}` : SERVICE_NAME}</title>
      </Head>
      <Header />
      <Box
        w="full"
        h="calc(100dvh - var(--chakra-sizes-16) - var(--chakra-sizes-16))"
        overflow="scroll"
      >
        <Box maxW={['5xl', '5xl', '84rem', '105rem']} mx="auto" my={20}>
          {children}
        </Box>
      </Box>
      <Footer />
    </>
  </Box>
);

export default CommonTemplate;
