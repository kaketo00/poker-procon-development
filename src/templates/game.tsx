import { Box } from '@chakra-ui/react';
import Head from 'next/head';
import { type PropsWithChildren } from 'react';

import Loading from '@/components/atoms/loading';
import Footer from '@/components/organisms/footer';
import Header from '@/components/organisms/header';
import { SERVICE_NAME } from '@/constants';
import { gameDeskHeight } from '@/styles';

interface Props extends PropsWithChildren {
  isLoading?: boolean;
  title?: string;
}

const GameTemplate = ({ isLoading = false, title, children }: Props) => (
  <Box
    w="full"
    minH="full"
    bgImage="url('/bg-game.png')"
    bgSize="cover"
    bgPos="center"
  >
    <>
      {isLoading && <Loading />}
      <Head>
        <title>{title ? `${title} - ${SERVICE_NAME}` : SERVICE_NAME}</title>
      </Head>
      <Header />
      <Box
        w={['full', 'auto', 'auto', 'full', 'full']}
        maxW={[
          '62.5rem', // 1,000px
          'full',
          'full',
          '96rem', // 1,536px
          '125rem', // 2,000px
        ]}
        minH={gameDeskHeight}
        my={[10, 10, 10, 10, 20]}
        mx={['auto', 20, 20, 'auto', 'auto']}
      >
        {children}
      </Box>
      <Footer />
    </>
  </Box>
);

export default GameTemplate;
