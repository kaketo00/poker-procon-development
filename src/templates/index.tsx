import CommonTemplate from './common';
import GameTemplate from './game';

export { CommonTemplate, GameTemplate };
