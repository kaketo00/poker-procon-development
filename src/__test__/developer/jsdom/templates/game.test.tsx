import { render } from '@testing-library/react';

import { GameTemplate } from '@/templates';

describe('GameTemplate', () => {
  test('スナップショット/default', () => {
    const { asFragment } = render(<GameTemplate />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/loading', () => {
    const { asFragment } = render(
      <GameTemplate title="ゲームログ" isLoading />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
