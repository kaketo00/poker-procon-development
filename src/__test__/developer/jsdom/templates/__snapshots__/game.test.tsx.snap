// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameTemplate スナップショット/default 1`] = `
<DocumentFragment>
  <div
    class="css-1r0he82"
  >
    <div
      class="css-1ir58rf"
    >
      <a
        href="/"
      >
        <h1
          class="chakra-heading css-zwnc2m"
        >
          Flush Code Hack
        </h1>
      </a>
    </div>
    <div
      class="css-em4281"
    />
    <div
      class="css-1phanch"
    >
      <p
        class="chakra-text css-bmni1x"
      >
        © Benesse i-Career Co., Ltd. All Rights Reserved.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GameTemplate スナップショット/loading 1`] = `
<DocumentFragment>
  <div
    class="css-1r0he82"
  >
    <div
      class="css-wty2ul"
    >
      <div
        class="chakra-spinner css-18kddkt"
      >
        <span
          class="css-8b45rq"
        >
          loading
        </span>
      </div>
      <p
        class="chakra-text css-8s549p"
      >
        読み込み中...
      </p>
    </div>
    <div
      class="css-1ir58rf"
    >
      <a
        href="/"
      >
        <h1
          class="chakra-heading css-zwnc2m"
        >
          Flush Code Hack
        </h1>
      </a>
    </div>
    <div
      class="css-em4281"
    />
    <div
      class="css-1phanch"
    >
      <p
        class="chakra-text css-bmni1x"
      >
        © Benesse i-Career Co., Ltd. All Rights Reserved.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
